File: BinaryTree/python/bottomRightValue.py
from collections import deque
from typing import Deque

from node import Node


def bottom_right_value(root: Node) -> int:
    queue: Deque[Node] = deque([root])
    current = root

    while queue:
        current = queue.popleft()
        if current.left:
            queue.append(current.left)
        if current.right:
            queue.append(current.right)
    return current.val

File: BinaryTree/csharp/BottomRightValue.cs
using System;
using System.Collections.Generic;

namespace BinaryTreeSolutions;

public static class BottomRightValue
{
    public static int Solve(TreeNode root)
    {
        if (root is null)
        {
            throw new ArgumentNullException(nameof(root));
        }

        var queue = new Queue<TreeNode>();
        queue.Enqueue(root);
        var current = root;

        while (queue.Count > 0)
        {
            current = queue.Dequeue();
            if (current.Left is not null) queue.Enqueue(current.Left);
            if (current.Right is not null) queue.Enqueue(current.Right);
        }

        return current.Val;
    }
}
